File: OOPS/Lab4/lab4.java
import java.util.Scanner;
import java.lang.*;

class Rectangle {
    float width, length;                
    class Centroid {
        float x, y;
        Centroid(float X, float Y){
            this.x = X;
            this.y = Y;
        }
        void show(){
            System.out.print(this.x + ',' + this.y + '\n');
        }
    };

    Rectangle (float l, float w){
        this.width = w;
        this.length = l;
    }
    void setLength(float l){
        this.length = l;
    }
    void setWidth(float w){
        this.width = w;
    }
    float perimeter(){
        return (this.length + this.width) * 2;
    }
    float area(){
        return this.length * this.width;
    }
    void show(){
        System.out.print("Length is: " + this.length + ", Width is: " + width + '\n');
        System.out.print("Diameter of Rect is: " + this.calcDiag() + '\n');
        System.out.print("Area of Rect is: " + this.area() + '\n');
        System.out.print("Perimeter of Rect is: " + this.perimeter() + '\n');
        Centroid tmp = this.centroid();
        System.out.print("Value of centroid is ");
        tmp.show();
    }
    boolean sameArea(Rectangle rect){
        if (this.area() == rect.area()){
            return true;
        }
        return false;
    }
    double calcDiag(){
        return Math.sqrt(this.length * this.length + this.width * this.width);
    }
    Centroid centroid(){
        return new Centroid(this.length/2, this.width/2);
    }

};

class lab4{

    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in); 
        float length, width;
        System.out.print("Enter Length and Width of Rectangle: ");
        length = sc.nextInt();
        width = sc.nextInt();
        Rectangle rect1 = new Rectangle(length, width);
        System.out.print("Enter Length and Width of Rectangle: ");
        length = sc.nextInt();
        width = sc.nextInt();
        Rectangle rect2 = new Rectangle(length, width);
        rect1.show();
        System.out.print("Diameter of Rect1 is: " + rect1.calcDiag() + "\n");
        System.out.print("Area of Rect1 is: " + rect1.area() + "\n");
        System.out.print("Perimeter of Rect1 is: " + rect1.perimeter() + "\n");
        rect2.show();
        System.out.print("Diameter of Rect2 is: " + rect2.calcDiag() + "\n");
        System.out.print("Area of Rect2 is: " + rect2.area() + "\n");
        System.out.print("Perimeter of Rect2 is: " + rect2.perimeter() + "\n");

        if(rect1.sameArea(rect2)){
            System.out.print("Rectangles have the Same Area\n");
        }
        else {
            System.out.print("Rectangles does not have the Same Area\n");
        }
    }
}

File: OOPS/Lab4/lab4.cpp
#include <iostream>
#include <cmath>

using namespace std;

class Rectangle {
    private:
        float width, length;                
        class Centroid {
            private:
                float x, y;
            public:
                Centroid(float X, float Y){
                    x = X;
                    y = Y;
                }
                void show(){
                    cout << x << ',' << y << endl;
                }
        };

    public:
        Rectangle (float l, float w){
            width = w;
            length = l;
        }
        void setLength(float l){
            length = l;
        }
        void setWidth(float w){
            width = w;
        }
        float perimeter(){
            return (length + width) * 2;
        }
        float area(){
            return length * width;
        }
        void show(){
            cout << "Length is: " << length << ", Width is: " << width << endl;
            cout << "Diameter of Rect is: " << calcDiag() << endl;
            cout << "Area of Rect is: " << area() << endl;
            cout << "Perimeter of Rect is: " << perimeter() << endl;
            Centroid *tmp = centroid();
            cout << "Value of centroid is ";
            tmp->show();
        }
        int sameArea(Rectangle rect){
            if (area() == rect.area()){
                return 1;
            }
            return 0;
        }
        float calcDiag(){
            return sqrt(length * length + width * width);
        }
        Centroid* centroid(){
            return new Centroid(length/2, width/2);
        }

};


int main(int argc, char const *argv[])
{
    float length, width;
    cout << "\nEnter Length and Width of Rectangle: ";
    cin >> length >> width;
    Rectangle rect1(length, width);
    cout << "Rectangle 1: \n";
    rect1.show();
    cout << "Rectangle 2: \n";
    cout << "\nEnter Length and Width of Rectangle: ";
    cin >> length >> width;
    Rectangle rect2(length, width);
    rect2.show();
    cout << endl;
    if(rect1.sameArea(rect2)){
        cout << "Rectangles have the Same Area\n";
    }
    else {
        cout << "Rectangles does not have the Same Area\n";
    }
    return 0;
}
